tests/tcam_mmio.h

```diff
@@ -0,0 +1,46 @@
+#ifndef __TCAM_MMIO_H__
+#define __TCAM_MMIO_H__
+
+#include <stdint.h>
+#include <stdio.h>
+#include "mmio.h"
+
+// TCAM register base and offsets
+#define TCAM_BASE      0x4000
```

Comment:
this should be set via function, I mean by default it should be 0x4000, otherwise there should be a method setting the TCAM_BASE, as it is going to be different based on the usecase and address map requirements.

Also, where's the decision of setting TCAM to 0x4000 came from, is it something you decided on your own?